crypto/fipsmodule/sha/keccak1600.c

```diff
@@ -7,65 +7,20 @@
  * https://www.openssl.org/source/license.html
  */
 
-#include <openssl/e_os2.h>
-#include <string.h>
 #include <assert.h>
 
-#if !defined(KECCAK1600_ASM)
-
-#if defined(__i386) || defined(__i386__) || defined(_M_IX86) || \
-    (defined(__x86_64) && !defined(__BMI__)) || defined(_M_X64) || \
-    defined(__mips) || defined(__riscv) || defined(__s390__) || \
-    defined(__EMSCRIPTEN__)
-/*
- * These don't have "and with complement" instruction, so minimize amount
- * of "not"-s. Implemented only in the [default] KECCAK_2X variant.
- */
-# define KECCAK_COMPLEMENTING_TRANSFORM
-#endif
-
 #if defined(__x86_64__) || defined(__aarch64__) || \
     defined(__mips64) || defined(__ia64) || \
     (defined(__VMS) && !defined(__vax))
-/*
- * These are available even in ILP32 flavours, but even then they are
- * capable of performing 64-bit operations as efficiently as in *P64.
- * Since it's not given that we can use sizeof(void *), just shunt it.
- */
+
+ // Facilitate ILP32 model, the above platforms are capable of performing
```

Comment:
I think the comment meaning has changed. I understand it's saying that `BIT_INTERLEAVE` is provided to support 32-bit-pointer platforms and that the computation there is still efficient (as efficient as) 64-bit platforms (didn't check how or why). I don't think the comment points to the above platforms. But I could be lacking understanding here.

Reply:
Yes, I may have misunderstood the meaning here. 
I understand that for _all except_ (corrected here) the above platforms the Interleave is defined by the length of the pointers. However, for the given platforms, the use of `sizeof(void *)` is avoided. I consider that even the platforms listed can perform in ILP32 efficiently, they still operate in P64 (64-bit) and have the interleave defined without check performed.
I can return the precious comment if it is not being clear here.

Reply:
I suggest returning the previous comment despite that it's still a bit blurry for me.